### PR TITLE
The new ES6 based Awesome books

### PR DESCRIPTION
**Pull request by Sami-Ullah-Tufail**
## ES6 Awesome Books 
Use of ES6 modules
Use of import and export functionality
Use of Local Storage

### CODE REVIEW

File: Modules/books.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+const booklist = document.getElementById('bookShelf');
+const author = document.getElementById('author');
+const book = document.getElementById('title');
+const addButton = document.getElementById('addButton');
+const books = JSON.parse(localStorage.getItem('books')) || [];
+/* eslint no-use-before-define: "error" */
+/* eslint-disable */
```
Comment (anagudelogu) [inline Modules/books.js]: - [x] Please avoid disabling linters as they are very useful to keep a consistent code format and ensure the use of best practices. Instead, please consider solving the errors! 🤓 🔧

File: Modules/books.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const bookname = book.value.trim();
+  const titler = author.value.trim();
+  // console.log(bookname);
+  // console.log(titler);
+  if (bookname !== '' && titler !== '') {
+    const newbook = { bookname, titler };
+    books.push(newbook);
+    storage();
+    renderbooks();
+    // console.log(books,newbook);
```
Comment (anagudelogu) [inline Modules/books.js]: - [x] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: Modules/nav.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const list = document.getElementById('list');
+const Newaddition = document.getElementById('addNew');
+const contact = document.getElementById('contact');
+
+// list.addEventListener('click', () => showsection('listSection'));
+// Newaddition.addEventListener('click', () => showsection('addSection'));
+// contact.addEventListener('click', () => showsection('contactSection'));
```
Comment (anagudelogu) [inline Modules/nav.js]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: Modules/books.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    bookStorage.textContent = `${newbook.bookname} by ${newbook.titler}`;
+
+    const removeButton = document.createElement('button');
+    removeButton.textContent = 'Remove';
+    removeButton.classList.add('remove-button');
+    removeButton.addEventListener('click', () => {
+      removeBook(index);
+    });
+
+    book_storage.appendChild(removeButton);
```
Comment (anagudelogu) [inline Modules/books.js]: - [ ] Testing your app, I found that adding a book isn't working:

[Screencast from 16-08-23 17:49:26.webm](https://github.com/Sami-Ullah-Tufail/Awesome-books-ES6/assets/77069011/00aa3071-3d44-41b4-a0df-f4eead362847)

Also, there's an error in the console about this `book_storage` variable. Please consider fixing this! 🤓 🔧

File: index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+Newaddition.addEventListener('click', () => showsection('addSection'));
+contact.addEventListener('click', () => showsection('contactSection'));
+
+addButton.addEventListener('click', () => addbooks());
+
+function initializeApp() {
+  renderbooks();
+}
+
+document.addEventListener('DOMContentLoaded', initializeApp);
```
Comment (anagudelogu) [inline index.js]: - [x] There's a missing requirement in this milestone:

image

Please consider implementing it! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @Sami-Ullah-Tufail, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of GitHub flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @Sami-Ullah-Tufail,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- All changes fixed ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**